### PR TITLE
187 color block

### PR DESCRIPTION
**Pull request by lovemecomputer**
Closes https://github.com/carbon-design-system/carbon-website/issues/187

adds properties to `<color-block>`:
- `showhex` toggles displaying of hex text
- `size` 
  - `size` takes values of `xs`, `s`, `m`, `l` (12px, 24px, 40px, 80px respectively)
  - `size` also accepts a number (e.g. `size="200"`, which would render a square that is 200px x 200px)
- leaving out these extra properties will default to no label, and a swatch size of 24px

**other changes:**
- updates .md pages to use the new properties
- updates documentation to explain properties & defaults
- renames things in `ColorCard` to avoid name conflicts

**not addressed in this PR**
- implementing support for carbon color tokens. additional Issue creature for this functionality at https://github.com/carbon-design-system/carbon-website/issues/288

### CODE REVIEW

File: src/components/ColorBlock/color-block.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.color-block {
+  position: relative;
+}
```
Comment (alisonjoseph) [inline src/components/ColorBlock/color-block.scss]:
```suggestion
  font-size: .875rem;
}
```

Comment (alisonjoseph) [reply src/components/ColorBlock/color-block.scss]: I think this should actually be a smaller font size. :D

Comment (lovemecomputer) [reply src/components/ColorBlock/color-block.scss]: cool, implemented!

**Comment by lovemecomputer** [comment]
alright, looks like all is good now pending design review! note, @alisonjoseph and i decided not to implement support of IBM color tokens yet as part of our current sprint. new issue for this functionality opened: https://github.com/carbon-design-system/carbon-website/issues/288

**Comment by alisonjoseph** [comment]
Cool! will merge in once design signs off

**Comment by claycrenshaw** [comment]
Looks awesome to me, @lovemecomputer!  One question: for "updates documentation to explain properties & defaults" can you point me to that documentation? Thanks.

**Pull request merged**